packages/cloudflare/src/cli/build/bundle-server.ts

```diff
@@ -109,11 +109,13 @@ fetch = globalThis.fetch;
 const CustomRequest = class extends globalThis.Request {
   constructor(input, init) {
     if (init) {
-      delete init.cache;
-      if (init.body?.__node_stream__ === true) {
+      init = {
+        ...init,
+        cache: undefined,
         // https://github.com/cloudflare/workerd/issues/2746
-        init.body = __cf_stream.Readable.toWeb(init.body);
-      }
+        // https://github.com/cloudflare/workerd/issues/3245
+        body: init.body instanceof __cf_stream.Readable ? ReadableStream.from(init.body) : init.body,
```

Comment:
Would this fail if `init.body` is a Readable from unenv ?

Reply:
From the top of my head I would say no.
It works with Node (/undici) only because a Readable is async iterable.
And we moved away from unenv Readable because it does not implement async iterator.

Reply:
I was thinking about user code actually, since this is overriding the global `Request`, user code or one of their lib could be using unenv or similar and `instanceof` would be false

Reply:
Ah, got it now.

There should be no more instances of unenv `Readable` after https://github.com/unjs/unenv/pull/363 (https://github.com/cloudflare/workerd/issues/3245 has some details). Before that it was still possible to create an unenv Readable via the http module.